addons/website_sale/models/sale_order.py

```diff
@@ -334,6 +334,10 @@ def _cart_accessories(self):
                         product not in products and
                         (not product.company_id or product.company_id == line.company_id) and
                         product._is_variant_possible(parent_combination=combination)
+                        and (
+                            product._get_contextual_price()
+                            or not self.website_id.prevent_zero_price_sale
```

Comment:
Reverse the condition, computing the price is way more costly than checking a field on the website.